codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -57,15 +57,25 @@
      * @param server {@code true} this is a server
      */
     public Http2FrameCodec(boolean server) {
-        this(server, new DefaultHttp2FrameWriter());
+        this(server, true);
+    }
+
+    /**
+     * Construct a new handler.
+     *
+     * @param server {@code true} this is a server
+     */
+    public Http2FrameCodec(boolean server, boolean verbose) {
+        this(server, verbose, new DefaultHttp2FrameWriter());
     }
 
     // Visible for testing
-    Http2FrameCodec(boolean server, Http2FrameWriter frameWriter) {
+    Http2FrameCodec(boolean server, boolean verbose, Http2FrameWriter frameWriter) {
         Http2Connection connection = new DefaultHttp2Connection(server);
-        frameWriter = new Http2OutboundFrameLogger(frameWriter, HTTP2_FRAME_LOGGER);
+        frameWriter = verbose ? new Http2OutboundFrameLogger(frameWriter, HTTP2_FRAME_LOGGER) : frameWriter;
         Http2ConnectionEncoder encoder = new DefaultHttp2ConnectionEncoder(connection, frameWriter);
-        Http2FrameReader reader = new Http2InboundFrameLogger(new DefaultHttp2FrameReader(), HTTP2_FRAME_LOGGER);
+        Http2FrameReader frameReader = new DefaultHttp2FrameReader();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "DefaultHttp2FrameReader". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2095)